client_files/ElasticSearch.sh

```diff
@@ -0,0 +1,167 @@
+yum -y install java-1.7.0-openjdk
+# Reference this for if we want to try JDK 8: http://tecadmin.net/install-java-8-on-centos-rhel-and-fedora/
+## wget --no-check-certificate --no-cookies --header "Cookie: oraclelicense=accept-securebackup-cookie" http://download.oracle.com/otn-pub/java/jdk/8u45-b14/jdk-8u45-linux-x64.rpm
+## rpm -ivh jdk-8u45-linux-x64.rpm
+
+# Verify JAVA is installed
+java -version
+
+# Set $JAVA_HOME
+#
+# http://askubuntu.com/questions/175514/how-to-set-java-home-for-openjdk
+#
+echo "JAVA_HOME=\"/usr/bin\"" >> /etc/environment
+source /etc/environment
+echo $JAVA_HOME 
+
+# Install Elasticsearch via yum repository
+#
+# https://www.elastic.co/guide/en/elasticsearch/reference/current/setup-repositories.html
+# 
+echo "******* Installing Elasticsearch *******"
+
+# Download and install the public signing key:
+cd ~/sources/meza1/client_files
+rpm --import https://packages.elastic.co/GPG-KEY-elasticsearch
+
+# Add yum repo file
+cd ~/sources/meza1/client_files
+cp ./elasticsearch.repo /etc/yum.repos.d/elasticsearch.repo
+
+# Install repo
+yum -y install elasticsearch
+
+# Configure Elasticsearch to automatically start during bootup
+echo "******* Adding Elasticsearch service *******"
+chkconfig --add elasticsearch
+
+# *** MANUAL INSTALLATION OPTION (delete) ***
+# cd ~/sources
+# curl -L -O https://download.elastic.co/elasticsearch/elasticsearch/elasticsearch-1.6.0.tar.gz
+# tar -xvf elasticsearch-1.6.0.tar.gz
+# cp -r elasticsearch-1.6.0 /etc/elasticsearch-1.6.0
+# cd /etc/elasticsearch-1.6.0/bin
+
+#
+# Elasticsearch Configuration
+#
+# https://www.elastic.co/guide/en/elasticsearch/reference/current/setup-configuration.html
+# 
+
+echo "******* Adding Elasticsearch configuration *******"
+# Add host name per https://github.com/elastic/elasticsearch/issues/6611
+echo "127.0.0.1 Meza1" >> /etc/hosts
+
+# Rename the standard config file and copy over our custom config file
+cd /etc/elasticsearch
+mv ./elasticsearch.yml ./elasticsearch-old.yml
+cd ~/sources/meza1/client_files
+cp ./elasticsearch.yml /etc/elasticsearch/elasticsearch.yml
+
+# Make directories called out in elasticsearch.yml
+# ref: http://elasticsearch-users.115913.n3.nabble.com/Elasticsearch-Not-Working-td4059398.html
+cd /var
+mkdir data
+cd data
+mkdir elasticsearch
+cd /var
+mkdir work
+cd work
+mkdir elasticsearch
+cd /var
+# Grant elasticsearch user ownership of these new directories
+chown -R elasticsearch /var/data/elasticsearch
+chown -R elasticsearch /var/work/elasticsearch
+
+#
+# Install Extension:Elastica and Extension:CirrusSearch
+#
+echo "******* Adding extensions to ExtensionLoader *******"
+cd ~/sources/meza1/client_files
+# Add Elastica and CirrusSearch to ExtensionSettings
+cp ~/sources/meza1/client_files/ExtensionSettingsElasticSearch.php /var/www/meza1/htdocs/wiki/ExtensionSettingsElasticSearch.php
+cat /var/www/meza1/htdocs/wiki/ExtensionSettingsElasticSearch.php >> /var/www/meza1/htdocs/wiki/ExtensionSettings.php
```

Comment:
No need to ```cp``` the file into .../wiki if you're just going to append its contents to ```ExtensionSettings.php```. These two lines can be condensed into one:

```bash 
cat /var/www/meza1/htdocs/wiki/ExtensionSettingsElasticSearch.php >> ./ExtensionSettingsElasticSearch.php
```